src/main/java/org/wildfly/security/authz/jacc/ElytronPolicyProvider.java

```diff
@@ -0,0 +1,192 @@
+    }
+
+    private PermissionCollection getPermissions(PermissionCollection staticPermissions, Principal[] principals) {
+        Permissions permissions = new Permissions();
+
+        addPermissions(permissions, staticPermissions);
+
+        try {
+            ElytronPolicyConfiguration policyConfiguration = ElytronPolicyConfigurationFactory.getCurrentPolicyConfiguration();
+
+            if (policyConfiguration != null) {
+                addPermissions(permissions, policyConfiguration.getUncheckedPermissions());
+
+                if (principals != null) {
+                    for (String roleName : getRoleMapper().getRoles(principals)) {
+                        Permissions rolePermissions = policyConfiguration.getRolePermissions().get(roleName);
+                        addPermissions(permissions, rolePermissions);
+                    }
+                }
+            }
+        } catch (PolicyContextException e) {
+            log.authzFailedGetDynamicPermissions(e);
+        }
+
+        return permissions;
+    }
+
+    private boolean impliesRolePermission(ProtectionDomain domain, Permission permission, ElytronPolicyConfiguration policyConfiguration) {
+        Set<String> roles = getRoleMapper().getRoles(domain.getPrincipals());
+
+        roles.add(ANY_AUTHENTICATED_USER_ROLE);
+
+        Map<String, Permissions> rolePermissions = policyConfiguration.getRolePermissions();
+
+        for (String roleName : roles) {
+            Permissions permissions = rolePermissions.get(roleName);
+
+            if (permissions != null) {
+                if (permissions.implies(permission)) {
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    private boolean impliesUncheckedPermission(Permission permission, ElytronPolicyConfiguration policyConfiguration) {
+        Permissions uncheckedPermissions = policyConfiguration.getUncheckedPermissions();
+
+        return uncheckedPermissions.implies(permission);
+    }
+
+    private boolean impliesExcludedPermission(Permission permission, ElytronPolicyConfiguration policyConfiguration) {
+        Permissions excludedPermissions = policyConfiguration.getExcludedPermissions();
+
+        return excludedPermissions.implies(permission);
+    }
+
+    private RoleMapper getRoleMapper() {
+        RoleMapper roleMapper;
+
+        try {
+            roleMapper = (RoleMapper) PolicyContext.getContext(RoleMapper.POLICY_CONTEXT_HANDLER_ID);
+        } catch (Exception e) {
+            throw log.authzCouldNotObtainRoleMapper(e);
+        }
+
+        if (roleMapper == null) {
+            throw log.authzRoleMapperNotRegistered();
+        }
+
+        return roleMapper;
+    }
+
+    private boolean supports(Permission permission) {
+        return permission.getClass().getPackage().equals(WebResourcePermission.class.getPackage());
```

Comment:
This check is not robust, because IIRC it is possible that getPackage() may return null for some class loaders.